packages/ag-charts-enterprise/src/features/zoom/zoomRange.ts

```diff
@@ -35,6 +38,35 @@ export class ZoomRange {
         return this.getRangeWithValues(this.initialStart, this.initialEnd);
     }
 
+    public extendToEnd(extent: number) {
+        return this.extendWith((end) => Number(end) - extent);
+    }
+
+    public extendWith(fn: (end: Date | number) => Date | number) {
+        if (!this.domain) return;
+
+        const end = this.domain.at(-1);
+        if (end == null) return;
+
+        this.dirty = true;
+
+        this.end = end;
```

Comment:
Discussed offline: Maybe add a comment that `this.dirty` maybe reset as a side-effect of setting these fields?